File: 1480. 1480-Running Sum of 1d Array/1480. 1480-Running Sum of 1d Array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */

//法一：不修改元素组，空间复杂度高
var runningSum = function(nums) {
    let res = []
    res.push(nums[0])
    for(let i = 0; i < nums.length - 1; i++){
        res.push(res[i]+nums[i+1])
    }
    return res
};

//法二：修改元素组，空间复杂度低
var runningSum = function(nums) {
    for(let i = 1; i < nums.length; i++){
        nums[i] = (nums[i-1]+nums[i])
    }
    return nums
};

File: 1480-1480-Running Sum of 1d Array/1480-1480-Running Sum of 1d Array.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */

//法一：不修改元素组，空间复杂度高
int* runningSum(int* nums, int numsSize, int* returnSize){
    int* res = malloc(numsSize * sizeof(int));
    * returnSize = numsSize;
    res[0] = nums[0];
    for(int i = 1; i < numsSize; i++){
        res[i] = res[i-1] + nums[i];
    }
    return res;
}

//法一：修改元素组，空间复杂度低
int* runningSum(int* nums, int numsSize, int* returnSize){
    * returnSize = numsSize;
    for(int i = 1; i < numsSize; i++){
        nums[i] = nums[i-1] + nums[i];
    }
    return nums;
}
